src/openms/source/ANALYSIS/ID/FIAMSDataProcessor.cpp

```diff
@@ -0,0 +1,265 @@
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2018.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+// 
+// --------------------------------------------------------------------------
+// $Maintainer: Timo Sachsenberg $
+// $Authors: Svetlana Kutuzova, Douglas McCloskey $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/ANALYSIS/ID/FIAMSDataProcessor.h>
+#include <OpenMS/ANALYSIS/ID/AccurateMassSearchEngine.h>
+#include <OpenMS/FORMAT/MzTabFile.h>
+#include <OpenMS/FILTERING/NOISEESTIMATION/SignalToNoiseEstimatorMedianRapid.h>
+#include <OpenMS/ANALYSIS/OPENSWATH/SpectrumAddition.h>
+
+
+namespace OpenMS {
+  /// default constructor
+  FIAMSDataProcessor::FIAMSDataProcessor() :
+      DefaultParamHandler("FIAMSDataProcessor"),
+      mzs_(),
+      bin_sizes_(),
+      sgfilter_(),
+      picker_()
+    {
+    defaults_.setValue("filename", "fiams", "The filename to use for naming the output files");
+    defaults_.setValue("dir_output", "", "The path to the directory where the output files will be placed");
+
+    defaults_.setValue("resolution", 120000.0, "The instrument settings: resolution");
+
+    defaults_.setValue("polarity", "positive", "The instrument settings: polarity");
+    defaults_.setValidStrings("polarity", ListUtils::create<String>("positive,negative"));
```

Comment:
for new code please use C++11 style:
`defaults_.setValidStrings("polarity", {"positive", "negative"});`